langchain4j-milvus/src/main/java/dev/langchain4j/store/embedding/milvus/MilvusEmbeddingStore.java

```diff
@@ -42,305 +47,326 @@
+                embeddingSearchRequest.maxResults(),
+                metricType,
+                consistencyLevel
+        );
+
+        SearchResultsWrapper resultsWrapper = CollectionOperationsExecutor.search(milvusClient, searchParam);
+
+        List<EmbeddingMatch<TextSegment>> matches = toEmbeddingMatches(
+                milvusClient,
+                resultsWrapper,
+                collectionName,
+                consistencyLevel,
+                retrieveEmbeddingsOnSearch
+        );
+
+        List<EmbeddingMatch<TextSegment>> result = matches.stream()
+                .filter(match -> match.score() >= embeddingSearchRequest.minScore())
+                .collect(toList());
+
+        return new EmbeddingSearchResult<>(result);
     }
 
-    /**
-     * @param token The token (API key) of the managed Milvus instance.
-     * @return builder
-     */
-    public Builder token(String token) {
-      this.token = token;
-      return this;
+    private void addInternal(String id, Embedding embedding, TextSegment textSegment) {
+        addAllInternal(
+                singletonList(id),
+                singletonList(embedding),
+                textSegment == null ? null : singletonList(textSegment)
+        );
     }
 
-    /**
-     * @param username The username. See details <a href="https://milvus.io/docs/authenticate.md">here</a>.
-     * @return builder
-     */
-    public Builder username(String username) {
-      this.username = username;
-      return this;
-    }
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> textSegments) {
+        List<InsertParam.Field> fields = new ArrayList<>();
+        fields.add(new InsertParam.Field(ID_FIELD_NAME, ids));
+        fields.add(new InsertParam.Field(TEXT_FIELD_NAME, toScalars(textSegments, ids.size())));
+        fields.add(new InsertParam.Field(METADATA_FIELD_NAME, toMetadataJsons(textSegments, ids.size())));
+        fields.add(new InsertParam.Field(VECTOR_FIELD_NAME, toVectors(embeddings)));
 
-    /**
-     * @param password The password. See details <a href="https://milvus.io/docs/authenticate.md">here</a>.
-     * @return builder
-     */
-    public Builder password(String password) {
-      this.password = password;
-      return this;
+        insert(this.milvusClient, this.collectionName, fields);
+        flush(this.milvusClient, this.collectionName);
     }
 
-    /**
-     * @param consistencyLevel The consistency level used by Milvus.
-     *                         Default value: EVENTUALLY.
-     * @return builder
-     */
-    public Builder consistencyLevel(ConsistencyLevelEnum consistencyLevel) {
-      this.consistencyLevel = consistencyLevel;
-      return this;
+    @Override
+    public void removeAll(Collection<String> ids) {
+        ensureNotEmpty(ids, "ids");
+        removeForVector(this.milvusClient, this.collectionName, format("%s in %s", ID_FIELD_NAME, toJson(ids).replace("\n", "")));
```

Comment:
can `MilvusMetadataFilterMapper.mapIn` be used here?

Reply:
totally agree

Reply:
Sorry, I checked carefully. mapIn is only valid for metadata filtering and is not applicable to filtering conditions based on primary keys.